lib/msal-common/src/client/ClientCredentialClient.ts

```diff
@@ -141,6 +141,8 @@ export class ClientCredentialClient extends BaseClient {
             };
         } else {
             const requestBody = this.createTokenRequestBody(request);
+            const queryParameters = this.createTokenQueryParameters(request);
+            const endpoint = StringUtils.isEmpty(queryParameters) ? authority.tokenEndpoint : `${authority.tokenEndpoint}?${queryParameters}`;
```

Comment:
There is a `UrlString` class where the logic of computing the new URL is better suited.

Reply:
[+1](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/9384b584a3383732d9070bd7ed345589f8969c3f/lib/msal-common/src/url/UrlString.ts#L88)